### ISSUE TITLE
v2 route file naming: route file convention not reading route folders correctly

### ISSUE THREAD

**Issue by lennerd** [remix-run/remix#5303]
### What version of Remix are you using?

1.11.1

### Are all your remix dependencies & dev-dependencies using the same version?

- [X] Yes

#### 2. Enable v2 routing convention

```
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  ignoredRouteFiles: ["**/.*"],
  future: {
    v2_routeConvention: true,
  },
};
```

#### 3. Create a route folder tree as shown below

```bash
$ tree app/routes
app/routes
├── _app
│   └── index.tsx
└── _app._index.tsx
```

### Expected Behavior

With the given folder tree above I would expect the output of `npx remix routes` to be as follows:

```
$ npx remix routes
<Routes>
  <Route file="root.tsx">
    <Route file="routes/_app/index.tsx">
      <Route index file="routes/_app._index.tsx" />
    </Route
  </Route>
</Routes>
```

`routes/_app/index.tsx`: a pathless route with `_app` defining a layout for all other routes prefixed with `_app`

`routes/_app._index.tsx`: an index route nested inside the layout route called `_app`

### Actual Behavior

Currently `npx remix routes` errors when given the folder tree above:

```bash
$ npx remix routes
Error: Path undefined defined by route "_app/index" conflicts with route "_app._index"
```

When you remove `_app/index.tsx` folder route, you can see the error:

```
$ npx remix routes
<Routes>
  <Route file="root.tsx">
    <Route index file="routes/_app/index.tsx" />
  </Route>
</Routes>
``` 

`routes/_app/index.tsx` should not be recognized as an index route.

**Comment by ctrlplusb** [comment]
I used the following structure to wrap the "root" route in a folder;

```
|- app
   |- routes
      |- _index
         |- index.tsx
```

i.e. `/app/routes/_index/index.tsx`

**Comment by lennerd** [comment]
@ctrlplusb what do you get, when you run `npx remox routes`?

**Comment by confix** [comment]
I guess `remix routes` produces the error.

Given the following layout

```
| - app
    | - routes
        | - test.tsx 
```

produces the following routes output

```
<Routes>
  <Route file="root.tsx">
    <Route path="est" file="outes/test.tsx" />
  </Route>
</Routes>
```

It seems like some regression removes the first character, which for nested routes without nested URLs removes the underscore and produces a route with path parameter.

Applied to your example, `_app/index.tsx` turns into `app/index.tsx` and produces a route for `/app`. `_app/_index.tsx` is turned into `app/_index.tsx`. Now you routes collide.

**Comment by mcansh** [comment]
@confix both of these issues are resolved in #5228 which should have landed in yesterdays release but were held back for unknown reasons

**Comment by confix** [comment]
@mcansh Ah, good to know. Gonna wait then. Thanks!

**Comment by MichaelDeBoey** [comment]
Closed by #5228
